src/dotnet-install.ps1

```diff
@@ -871,7 +871,9 @@ if (!$isAssetInstalled) {
     $isAssetInstalled = Is-Dotnet-Package-Installed -InstallRoot $InstallRoot -RelativePathToPackage $dotnetPackageRelativePath -SpecificVersion $SpecificVersion
 }
 
+# Version verification failed. More likely something is wrong either with the downloaded content or with the verification algorithm.
 if (!$isAssetInstalled) {
+    Say-Error "Failed to verify the version of installed `"$assetName`", installation source: $DownloadLink, installation location: $InstallRoot. Report the bug at https://github.com/dotnet/install-scripts/issues."
```

Comment:
Nit: A new line before `installation source: ...` and `install location: ...` could make this easier to read. 

Otherwise, looks good to me.